num_enum/src/lib.rs

```diff
@@ -28,6 +29,13 @@ pub trait TryFromPrimitive: Sized {
     fn try_from_primitive(number: Self::Primitive) -> Result<Self, Self::Error>;
 }
 
+pub trait ConstTryFromPrimitive: Sized {
```

Comment:
How would you feel about making `ConstTryFromPrimitive` a subtrait of `TryFromPrimitive` (and similarly for the others)? I think that would allow us to remove `ConstTryFromPrimitiveError` (we could just use `TryFromPrimitiveError` as the bound would still hold), and would mean the new const generic on `EnumInfo` could disappear?

I guess it's a bit annoying to have to derive both if you just want the const version, but I'm also not sure how common it would be to only want the const version, and also it mirrors how things like `PartialEq`/`Eq` work?

Reply:
This sounds great! It reminds me of what I was trying to do with `ConstantTimePartialOrd` in `subtle` a while ago (https://github.com/dalek-cryptography/subtle/pull/98). I'll see if this works.

Reply:
I personally do not have a use case for const-only derives -- I think I was a little confused about how `const fn` can also be a normal runtime `fn` and how to separate those. I think a subtrait and merging the errors (which I would definitely like to do) would be ideal!